File: mobile/lib/models/product.dart
class Product {
  final int? id;
  final String name;
  final String? barcode;
  final double calories;
  final double protein;
  final double fat;
  final double carbs;
  final String? verdict;
  final String? verdictText;
  final bool verdictIsMock;

  Product({
    this.id,
    required this.name,
    this.barcode,
    required this.calories,
    required this.protein,
    required this.fat,
    required this.carbs,
    this.verdict,
    this.verdictText,
    this.verdictIsMock = false,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'] ?? '',
      barcode: json['barcode'],
      calories: (json['calories'] as num).toDouble(),
      protein: (json['protein'] as num).toDouble(),
      fat: (json['fat'] as num).toDouble(),
      carbs: (json['carbs'] as num).toDouble(),
      verdict: json['verdict'],
      verdictText: json['verdict_text'],
      verdictIsMock: json['verdict_is_mock'] ?? false,
    );
  }
}

File: backend/app/schemas/product.py
from typing import Optional
from pydantic import BaseModel


class ProductSchema(BaseModel):
    id: int
    barcode: str | None
    name: str
    calories: float
    protein: float
    fat: float
    carbs: float
    composition: str | None
    is_verified: bool

    model_config = {"from_attributes": True}


class ProductWithVerdict(ProductSchema):
    """ProductSchema + опциональный AI-вердикт (используется в /barcode и /correct-product)."""
    verdict: Optional[str] = None
    verdict_text: Optional[str] = None
    verdict_is_mock: bool = False


class ProductCreate(BaseModel):
    name: str
    barcode: str | None = None
    calories: float
    protein: float
    fat: float
    carbs: float
    composition: str | None = None
